atlasdb-jdbc/src/main/java/com/palantir/atlasdb/keyvalue/jdbc/JdbcKeyValueService.java

```diff
@@ -278,13 +283,15 @@ public Void apply(DSLContext ctx) {
         if (timestampByCell.isEmpty()) {
             return ImmutableMap.of();
         }
-        return run(new Function<DSLContext, Map<Cell, Value>>() {
-            @Override
-            public Map<Cell, Value> apply(DSLContext ctx) {
+
+        Map<Cell, Value> toReturn = new HashMap<>();
+
+        for (List<Entry<Cell, Long>> parition : Iterables.partition(timestampByCell.entrySet(), batchSizeForReads)) {
+            toReturn.putAll(run(ctx -> {
                 Select<? extends Record> query = getLatestTimestampQueryManyTimestamps(
                         ctx,
                         tableRef,
-                        toRows(timestampByCell));
```

Comment:
Do we care about the other usages of toRows?  They aren't batched at the moment.